pkg/ovhsdk/api/services.go

```diff
@@ -0,0 +1,87 @@
+package api
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/ovh/go-ovh/ovh"
+)
+
+// Pattern : https://github.com/tmrts/go-patterns/blob/master/idiom/functional-options.md
+type Options struct {
+	ResourceName string
+	OrderBy      string
+	Routes       string
+	Sort         string
+}
+
+type Option func(*Options)
+
+func ResourceName(resourceName string) Option {
```

Comment:
why all of this complexity just to create a struct with nullable fields?
```
type Options struct {
	ResourceName *string
	OrderBy      *string
	Routes       *string
	Sort         *string
}
```

	// Checking if a field is set
	if opts.ResourceName != nil {
		fmt.Println(*opts.ResourceName)
	} else {
		fmt.Println("ResourceName is not set")
	}